custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js

```diff
@@ -13,7 +13,9 @@ function MonthModalController($location, $uibModalInstance, dateHelperService) {
 
     var startYear = dateHelperService.getStartingYear(isSDD);
 
-
+    var currentDate = new Date();
+    var maxYear = dateHelperService.isBetweenFirstAndThirdDayOfCurrentMonth(currentDate) ?
+        dateHelperService.getPreviousMonthDate().getFullYear() : currentDate.getFullYear();
```

Comment:
What do you think about combining `dateHelperService.isBetweenFirstAndThirdDayOfCurrentMonth` and `dateHelperService.getPreviousMonthDate` to a function that is something like `dateHelperService.getDefaultDate` (not sure on the name) to avoid repeating this duplicate logic in several places?

Reply:
That is definitely better. Changing it. 
(Please mention if you have any name suggestions :stuck_out_tongue_winking_eye: )